### ISSUE TITLE
Fail when `npm version` fails

### ISSUE THREAD

**Issue by LuudJanssen** [release-it/release-it#765]
First of all: thank you! This amazing tool helped us save a lot of time 🙌

I was often getting the following error:

```
failed to stream package from json: unhandled input: Cannot publish over existing version.
```

I checked the semantic versioning logic and even tried `release-it <version>` to make sure it was trying to publish the right version. After some debugging with the `-VV` flag I noticed that my `preversion` script was failing, which let to `npm version` never being executed.

IMO if the `npm version` command fails, the release should also fail, because it can never successfully upload a new version to NPM.

To reproduce this you could create a project with the following `preversion` script in your `package.json`:

```json
{
  "preversion": "exit 1;"
}
```

This should always fail and lead to `release-it` trying to re-submit the same version to NPM.

**Comment by webpro** [comment]
> IMO if the `npm version` command fails, the release should also fail, because it can never successfully upload a new version to NPM.

Yes, under normal circumstances it should fail. But not always, since there is a `--no-increment` option which might be combined with a `--npm.publish` (e.g. when drafting a GitHub release first and then publishing this release together with the npm package).

That's why it only a warning now. Will look into this, thanks for the report.

**Comment by webpro** [comment]
This change is released in [v14.7.0](https://github.com/release-it/release-it/releases/tag/14.7.0), halting the process when `npm version` fails.
